internal/build/phase_test.go

```diff
@@ -139,17 +163,18 @@ func testPhase(t *testing.T, when spec.G, it spec.S) {
 
 			when("app is a dir", func() {
 				it("preserves original mod times", func() {
-					assertAppModTimePreserved(t, subject, &outBuf, &errBuf)
+					assertAppModTimePreserved(t, lifecycle, phaseFactory, &outBuf, &errBuf)
 				})
 			})
 
 			when("app is a zip", func() {
 				it("preserves original mod times", func() {
 					var err error
-					subject, err = CreateFakeLifecycle(filepath.Join("testdata", "fake-app.zip"), docker, logger)
+					lifecycle, err = CreateFakeLifecycle(filepath.Join("testdata", "fake-app.zip"), docker, logger, repoName)
```

Comment:
I've long held a belief that there's a go idiom that the logger should be the final argument passed to a method, but I can't currently find any evidence to back that up. It would feel more consistent with other parts of our codebase.

Reply:
My $0.02, I would think it should be near the beginning because it's a **common** argument, similar to context in the standard lib.

As a visual example:

```go
func foo1(context, logger, arg0, arg1, arg2)
func foo2(context, logger, arg0, arg1)
func bar(context, arg0, arg1)
func fizz(arg0, arg1)
```

vs 


```go
func foo1(context, arg0, arg1, arg2, logger)
func foo2(context, arg0, arg1, logger)
func bar(context, arg0, arg1)
func fizz(arg0, arg1)
```

Reply:
The closest sister method to `CreateFakeLifecycle` is `build.NewLifecycle(docker, logger)` so I would like to keep `docker, logger` as either the first two or the last two arguments. Going with the first two because ¯\_(ツ)_/¯